hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SequenceIdGenerator.java

```diff
@@ -416,17 +436,13 @@ public static void upgradeToSequenceId(SCMMetadataStore scmMetadataStore)
         while (iterator.hasNext()) {
           X509Certificate cert =
               iterator.next().getValue().getX509Certificate();
-          if (HASecurityUtils.isSelfSignedCertificate(cert) &&
-              HASecurityUtils.isCACertificate(cert)) {
-            largestRootCertId =
-                Long.max(cert.getSerialNumber().longValueExact(),
-                    largestRootCertId);
-          }
+          largestCertId = Long.max(
```

Comment:
Is it possible here that we have an older certificate that still has an id out of the long range here?

Reply:
@Galsza , do you mean a serial ID which is beyond the scope of LONG can represent?

Reply:
Yes, that's what I mean. In some other place we have encountered such an issue before.

Reply:
Long.MAX_VALUE is a very big data.  If every 1s it generate 1000 certificates, then Long.MAX_VALUE will be exhausted by 292 million years.  For a fresh new Ozone cluster, which certificate serial ID starts from 1, Long.MAX_VALUE is enough for this cluster's whole lifetime.  For a already existing Ozone cluster, it's existing max certificate serial ID could be a bigger number. But given the certificate rotation is yearly scheduled, it's very unlikely it will exhaust the numbers.

Reply:
as we used nanotime at one point in the execution before, and as the current nanotime is somewhere around 1.7*10^18, while Long.MAX_VALUE is somewhere around 9.2*10^18 we are still fairly safe to exhaust this counter fairly long time from now, I agree :)